src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php

```diff
@@ -0,0 +1,256 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\PhpUnit;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\DocBlock\DocBlock;
+use PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Indicator\PhpUnitTestCaseIndicator;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+/**
+ * @author Gert de Pagter
+ */
+final class PhpUnitTestAnnotationFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
+{
+    private $callback;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        $this->callback = $this->configuration['annotation'] ? 'applyTestAnnotation' : 'removeTestAnnotation';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Adds or removes @test annotations from tests, following configuration.',
+            [
+                new CodeSample("<?php
+class Test extends \\PhpUnit\\FrameWork\\TestCase
+{
+public function testItDoesSomething() {}}\n"),
+
+                new CodeSample("<?php
+class Test extends \\PhpUnit\\FrameWork\\TestCase
+{
+    /**
+     * @test
+     */
+    public function itDoesSomething() {} }\n", ['annotation' => false])
```

Comment:
@BackEndTea 

Just an idea (since you and I have seen this in at least one repository):

* some people use `camelCase` with `@test` annotation
* some people use `snake_case` with `@test` annotation
* some people use prefix `test` followed by `PascalCase`
* some maybe even use prefix `test` followed by `PascalCase` and also have a redundant `@test` annotation

Not sure if the fixer should take care of the last case (we could just add a separate fixer), but I think it would be awesome if the fixer could cater to all the 3 cases before, what do you think?